go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -70,16 +74,72 @@ type serviceConfig struct {
 	service *kapi.Service
 	// Were those rules etp:local + Host Networked rules
 	etpHostRules bool
+	// Track etp:local + Non Host Networked rules. Used only in LGW.
+	etpLocalNonHostRules bool
+}
+
+// isHostNetworkedService returns etpHostRules & etpLocalNonHostRules booleans
+// etpHostRules is true if provided service has host-networked endpoints (given endpoints exist for the service)
+// etpLocalNonHostRules is true if provided service is of type etp=local & gwmode=local & endpoints are not host-networked
+func (npw *nodePortWatcher) isHostNetworkedService(service *kapi.Service) (bool, bool) {
```

Comment:
is it possible to not use two different variables to declare host rules and non host rules depending on gw mode? I mean just return if the service has host network endpoints, then whatever consumes this function decides what to do based on  considering gateway mode?